### PR TITLE
Feature/hyperparameter sweep

### PR DESCRIPTION
**Pull request by SmillaDue**
W&B Hyperparameter Sweep Infrastructure:

Added scripts and configs to enable hyperparameter sweeps on GCP Vertex AI.
Integrated W&B sweep configuration and Docker support for scalable experiments.
Configuration and Model Updates:

Updated and added multiple config files for model training and sweeps.
Improved model implementation in model.py.
Added GPU support and platform-specific PyTorch dependencies.
Experiment Results and Tracking:

Added new sweep results and configuration files under wandb/.
Included new sweep_densenet.yaml in configs.
Fixed environment variable handling for GCP sweep jobs.

Ran final model, and updated final model parameters: 
lr: 0.0001
weight decay: 0.001
dropout: 0.1 
batchsize: 32
epochs: 10

### CODE REVIEW

**Comment by SmillaDue** [comment]
I can't make the ubuntu checks work for some reason. Don't have anymore time to look at it now. However, the code works and the other tests go through. The ubuntu fail is some weird thing with the initialisation of pytorch. I don't think it is critical - so maybe a merge is okay
